### ISSUE TITLE
tsconfig.json incorrectly relies upon browser DOM types

### ISSUE THREAD

**Issue by ericdrobinson** [AdobeXD/typings#65]
UXP supports a [specific _subset_](https://adobexdplatform.com/plugin-docs/reference/uxp/ui-index.html) of browser DOM APIs. [TypeScript's DOM lib](https://github.com/microsoft/TypeScript/blob/master/src/lib/dom.generated.d.ts), on the other hand, is _extremely_ comprehensive. Configuring an environment with TypeScript's built-in DOM library leads to [confusing bugs](https://github.com/AdobeXD/typings/issues/64) and causes tools to incorrectly suggest types.

Recommendation:

1. Remove the `dom` entry from the `tsconfig.json` `lib` array.
1. Create UXP-specific type declaration files for the following:
    1. [UI Classes](https://adobexdplatform.com/plugin-docs/reference/uxp/ui-classes-index.html) (e.g. [`Attr`](https://adobexdplatform.com/plugin-docs/reference/uxp/class/Attr.html) and friends)
    1. [HTML Elements](https://adobexdplatform.com/plugin-docs/reference/uxp/ui-html-index.html) (e.g. [`HTMLAnchorElement`](https://adobexdplatform.com/plugin-docs/reference/uxp/class/HTMLAnchorElement.html) and friends)
    1. [Events](https://adobexdplatform.com/plugin-docs/reference/uxp/events-index.html) (e.g. [`BaseUIEvent`](https://adobexdplatform.com/plugin-docs/reference/uxp/class/BaseUIEvent.html) and friends)

As Adobe incorporates more HTML and standard browser DOM APIs into UXP, the type declaration files can be updated to reflect the enhancements. This will help resolve an entire class of typo/error/etc.

**Comment by pklaschka** [comment]
I'm currently unable to update the repo as I'm in the middle of exams and don't have enough time. Until August 8th, I'll review (and merge) pull requests, view comments, etc., but please understand that I don't have the time to actively update the repo for these next two weeks. Thank you very much for your understanding 🙂

**Comment by pklaschka** [comment]
I agree that this would be ideal. However, this would require a lot of manual labor and, at least for me, is not feasible to do manually. Also, as Adobe only points to the MDN docs, it would be even more time-consuming to provide fully documented typings.

All in all, while I agree that it would be better to do so, I currently don't think it's feasible to maintain it that way. If anyone would be willing to create and maintain this aspect of the typings, feel more than welcome to do so, but personally, I won't have the time to do that...

@ashryanbeats @kerrishotts @peterflynn Any chance of getting some sort of structured format about all the classes and methods of UXP? Generating typings from that could (perhaps) make this idea feasible, and I assume it exists anyway (you haven't written this part of the docs manually?)...
